packages/devtools-reps/src/reps/tests/promise.js

```diff
@@ -0,0 +1,208 @@
+  });
+
+  it("should render as expected", () => {
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "fulfilled" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test Promise fulfilled with object", () => {
+  const object = stubs.get("FulfilledWithObject");
+  const defaultOutput = `Promise { <state>: "fulfilled", <value>: Object }`;
+
+  it("correctly selects PromiseRep Rep for Promise fulfilled with an object", () => {
+    expect(getRep(object)).toBe(PromiseRep.rep);
+  });
+
+  it("should render as expected", () => {
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "fulfilled" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test Promise fulfilled with array", () => {
+  const object = stubs.get("FulfilledWithArray");
+  const defaultOutput = `Promise { <state>: "fulfilled", <value>: [3] }`;
+
+  it("correctly selects PromiseRep Rep for Promise fulfilled with an array", () => {
+    expect(getRep(object)).toBe(PromiseRep.rep);
+  });
+
+  it("should render as expected", () => {
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "fulfilled" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test on DOM node MouseOver", () => {
+  const stub = stubs.get("FulfilledWithNode");
+  const grips = getSelectableInInspectorGrips(stub);
+
+  it("has one node grip", () => {
+    expect(grips.length).toBe(1);
+  });
+
+  it("calls with expected argument when mouseover is fired on the node element", () => {
+    let mouseOverValue;
+    const onDOMNodeMouseOver = (object) => {
+      mouseOverValue = object;
+    };
+
+    const renderedComponent = shallow(Rep({
+      object: stub,
+      onDOMNodeMouseOver,
+    }));
+
+    renderedComponent.find(".objectBox-node").simulate("mouseover");
+    expect(mouseOverValue).toEqual(grips[0]);
+  });
+});
+
+describe("test on DOM node MouseOut", () => {
+  const stub = stubs.get("FulfilledWithNode");
+  const grips = getSelectableInInspectorGrips(stub);
+
+  it("has one node grip", () => {
+    expect(grips.length).toBe(1);
+  });
+
+  it("calls with expected argument when mouseout is fired on the node", () => {
+    let called = false;
+    const onDOMNodeMouseOut = (object) => {
```

Comment:
Could we use `jest.fn()` like I suggested for the `onDomMouseOver` test ?
It could be something like: 
```js
it("calls the expected function on mouseout", () => {
  const onDOMNodeMouseOut = jest.fn();
  const wrapper = renderRep(object, {onDOMNodeMouseOut});

  wrapper.simulate("mouseout");

  expect(onDOMNodeMouseOut.mock.calls.length).toBe(1);
});
```